### ISSUE TITLE
Inconsistent data at /stats

### ISSUE THREAD

**Issue by ghost** [searxng/searxng#2650]
**Version of SearXNG, commit number if you are using on master branch and stipulate if you forked SearXNG**
2256ba2ff

**How did you install SearXNG?**
Docker

**What happened?**
When refreshing `/stats` I get different results: they randomly change between two pages.

**How To Reproduce**
Make a few queries, go to stats, refresh multiple times (can take up to 5 refreshes for me), and see different (older) stats

**Expected behavior**
Newest stats displayed, or, perhaps, a consistent version of them

**Screenshots & Logs**
These ones were taken with no queries made in between them!
![image](https://github.com/searxng/searxng/assets/59206807/a4a5ecb7-e047-481d-93a9-1d2404c23843)
![image](https://github.com/searxng/searxng/assets/59206807/be099b48-4381-40be-8928-58f9eb6bc64e)

**Additional context**
-

**Comment by unixfox** [comment]
Please explain a bit more which tutorial you used to install searxng.

**Comment by ghost** [comment]
https://docs.searxng.org/admin/installation-docker.html
I used your official image (searxng/searxng)

**Comment by return42** [comment]
> When refreshing `/stats` I get different results: they randomly change between two pages.

Stats are stored in the workers / I assume you have two CPU :-)

https://github.com/searxng/searxng/blob/9100a48541f469df2973d3d42e8ed8f6bba4fac1/dockerfiles/uwsgi.ini#L6-L7

**Comment by ghost** [comment]
Why are there only 2 different stats pages, if I have 12 workers (because 12 cores)
I guess it is a coincidence then?

**Comment by return42** [comment]
> I guess it is a coincidence then?

Yes, AFAIK uWSGI selects the worker by its own rules (balancing?) .. in your case may be only two of n are used .. 

@dalf can you confirm, that stats data are stored in the walker / I refer to this..

https://github.com/searxng/searxng/blob/9100a48541f469df2973d3d42e8ed8f6bba4fac1/searx/metrics/__init__.py#L33-L34

https://github.com/searxng/searxng/blob/9100a48541f469df2973d3d42e8ed8f6bba4fac1/searx/metrics/__init__.py#L72-L80

https://github.com/searxng/searxng/blob/9100a48541f469df2973d3d42e8ed8f6bba4fac1/searx/metrics/models.py#L114-L117

https://github.com/searxng/searxng/blob/9100a48541f469df2973d3d42e8ed8f6bba4fac1/searx/metrics/models.py#L148-L150

**Comment by ghost** [comment]
In case it is stored on the workers and you don't fix it, I believe that it should be written in docs to avoid further questions.
Also, `only two of n are used` - I definitely should reduce it to avoid unnecessary overhead )

**Comment by dalf** [comment]
> @dalf can you confirm, that stats data are stored in the walker

Yes

> Why are there only 2 different stats pages, if I have 12 workers (because 12 cores) I guess it is a coincidence then?

According to my experience, uwsgi dispatches the requests the same worker as long it is available.

> In case it is stored on the workers and you don't fix it

FYI, there are already two implementations of the storage: if the [metrics are disabled in settings.yml](https://github.com/searxng/searxng/blob/9100a48541f469df2973d3d42e8ed8f6bba4fac1/searx/settings.yml#L14), the code [uses two alternative Python classes](https://github.com/searxng/searxng/blob/9100a48541f469df2973d3d42e8ed8f6bba4fac1/searx/metrics/models.py#L160C7-L167). A third implementation has been discussed : store the values in Redis.

**Comment by ghost** [comment]
> According to my experience, uwsgi dispatches the requests the same worker as long it is available.
Not sure, there weren't any other requests and I doubt that 3 refreshes would make it too busy.

Well, until Redis implementation is used, shouldn't you make note of it on `/stats` page or in the docs?

**Comment by return42** [comment]
> shouldn't you make note of it on /stats page or in the docs?

Nope .. we prefer to implement a persistent storage solution one day.
